modules/gapi/include/opencv2/gapi/streaming/onevpl/onevpl_data_provider_interface.hpp

```diff
@@ -0,0 +1,42 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2021 Intel Corporation
+
+#ifndef GAPI_STREAMING_ONEVPL_ONEVPL_DATA_PROVIDER_INTERFACE_HPP
+#define GAPI_STREAMING_ONEVPL_ONEVPL_DATA_PROVIDER_INTERFACE_HPP
+#include <exception>
+#include <string>
+
+#include <opencv2/gapi/own/exports.hpp> // GAPI_EXPORTS
+
+namespace cv {
+namespace gapi {
+namespace wip {
+
+struct GAPI_EXPORTS DataProviderException : public std::exception {
+    virtual ~DataProviderException() {};
+};
+
+struct GAPI_EXPORTS DataProviderSystemErrorException : public DataProviderException {
+    DataProviderSystemErrorException(int error_code, const std::string& desription = std::string());
+    virtual ~DataProviderSystemErrorException();
+    virtual const char* what() const noexcept override;
+
+private:
+    std::string reason;
+};
```

Comment:
y u need a special exception type here? usually we throw a `logic_error` and its string argument is enough to provide `reason` like here. I see there's also error_code, but it looks discarded right now

Reply:
in recognition reason: i believe user can recognize that and make  a proper decision on explicit error.

BTW it depict explicit intention during DataProvider implementation .When i see an `abstract interface` i ask myself: how i should process an unrecoverable error here: `logic_error`  or `runtime_error` or `bad_alloc` of `filesystem` error? Answer is `DataProviderSystemErrorException `

Reply:
We don't have a project-wide exception / error-handling policy here (except the one I mentioned above), so I'm just wondering whether it is the right time to introduce / derive a new one or keep the old approach (before the one is introduced)